File: 2485. Find the Pivot Integer/find-the-pivot-integer.java
class Solution {
    public int pivotInteger(int n) {
        for(int i = 1; i <= n; i++){
            int leftsum = (i * (i+1))/2; 
            int rightsum = (n * (n+1))/2 - (i * (i-1))/2;

            if(leftsum == rightsum){
                return i;
            };
        };
        return -1;
    }
}

File: 2485. Find the Pivot Integer/find-the-pivot-integer.rs
impl Solution {
    pub fn pivot_integer(n: i32) -> i32 {
        let mut leftsum : i32 = 0;
        let mut rightsum : i32 = 0;

        for i in 0..=n {
            leftsum = (i * (i + 1)) / 2;
            rightsum = (n * (n + 1)) / 2 - (i * (i - 1)) / 2;

            if(leftsum == rightsum){
                return i;
            };
        };
        return -1;
    }
}